File: back-end/src/main/java/com/trucks/application/port/PortRepository.java
package com.trucks.application.port;

import java.util.List;
import java.util.UUID;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import com.trucks.domain.entity.Port;

public interface PortRepository {

    Port save(Port port);

    Optional<Port> find(UUID id);

    Page<Port> findAll(PageRequest pageRequest);

    Port update(UUID id, Port port);
    
    Port delete(UUID id);

    List<Port> deleteAll();
}

File: front-end/src/application/repository/portRepository.ts
import { Port } from "../../domain/entity/port";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from "../../utils/ull-uuid";

export interface PortRepository {
    save: (port: Port) => Promise<Either<DataError, Port>>;
    
    find: (id: UllUUID) => Promise<Either<DataError, Port>>;
    
    findAll: (pageRequest: any) => Promise<Either<DataError, Port[]>>;
    
    update: (id: UllUUID, port: Port) => Promise<Either<DataError, Port>>;
    
    delete: (id: UllUUID) => Promise<Either<DataError, Port>>;
    
    deleteAll: () => Promise<Either<DataError, Port[]>>;
}